From ce52cab608146044f68dee2da30da003c6ed0be7 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Mon, 06 May 2019 17:10:44 +0300
Subject: [PATCH] [SW-0] goya/gaudi: pll: remove cpu pll handling

Remove completely the CPU PLL handling from preboot as we cannot
change it while running on it.
The only one that should change/setup the CPU PLL is BTL.
For the JTAG case, we should update the script, which loads the
preboot/U-Boot directly to SRAM, to setup the CPU PLL if needed.

Change-Id: I659e921fa9e6bea1ff4bced5cd2416a855af4e7e
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/common/pll.c b/common/pll.c
index b365c52..214842d 100644
--- a/common/pll.c
+++ b/common/pll.c
@@ -195,7 +195,6 @@
 {
 	int i, pll_num;
 	struct hl_pll_cfg *hl_pll_cfg;
-	enum hl_pll_bypass_val bval;
 
 	/*
 	 * Note: we assume CPU frequency is not changed while CPU is running,
@@ -210,13 +209,6 @@
 	*pll_partial_init = hl_pll_partial_init_get();
 
 	for (i = 0; i < pll_num; i++) {
-		/* DIV0 is the undivided PLL output */
-		bval = hl_pll_bypass_get(hl_pll_cfg[i].base, HL_PLL_DIV0);
-		if (hl_pll_cfg[i].base == hl_pll_get_cpu_pll_base_off() &&
-		    bval != HL_PLL_REF_CLK &&
-		    bval != HL_PLL_DIV_REF)
-			continue;
-
 		if (*pll_partial_init)
 			hl_pll_bypass(hl_pll_cfg + i, false);
 		else
diff --git a/common/pll.h b/common/pll.h
index e1ccf34..8845630 100644
--- a/common/pll.h
+++ b/common/pll.h
@@ -62,7 +62,6 @@
 int hl_pll_init(unsigned int *pll_partial_init);
 int hl_pll_get_cfg(struct hl_pll_cfg **hl_pll_cfg);
 u64 hl_pll_get_cpu_cycles_per_usec(void);
-u64 hl_pll_get_cpu_pll_base_off(void);
 u32 hl_pll_get_freq(u64 pll_base, enum hl_pll_div_out div);
 
 int hl_pll_partial_init_get(void);
diff --git a/gaudi/asic.c b/gaudi/asic.c
index a4aefae..c540cf3 100644
--- a/gaudi/asic.c
+++ b/gaudi/asic.c
@@ -15,15 +15,6 @@
 
 static struct hl_pll_cfg gaudi_pll_cfg[] = {
 	{
-		.name = "CPU", .base = mmPSOC_CPU_PLL_BASE,
-		.nr = 0, .nf = 47, .od = 1, .nb = 23, /* 1200 */
-		.div_factor = { 0, 1, 47, -1 },
-		.div_sel = { HL_PLL_PLL_CLK,
-			     HL_PLL_DIV_PLL,
-			     HL_PLL_DIV_PLL,
-			     HL_PLL_REF_CLK
-		},
-	}, {
 		.name = "DMA_W", .base = mmDMA_W_PLL_BASE,
 		.nr = 0, .nf = 59, .od = 1, .nb = 29, /* 1500 */
 		.div_factor = { 0, 1, -1, -1 },
diff --git a/gaudi/common.c b/gaudi/common.c
index e7c2324..90d960a 100644
--- a/gaudi/common.c
+++ b/gaudi/common.c
@@ -44,11 +44,6 @@
 	return mmPSOC_GLOBAL_CONF_COLD_RST_FLOPS_3;
 }
 
-u64 hl_pll_get_cpu_pll_base_off(void)
-{
-	return mmPSOC_CPU_PLL_BASE;
-}
-
 int hl_pll_partial_init_get(void)
 {
 	return regmap_readl(mmPSOC_GLOBAL_CONF_COLD_RST_FLOPS_2);
diff --git a/goya/asic.c b/goya/asic.c
index 2fdc652..d1c0add 100644
--- a/goya/asic.c
+++ b/goya/asic.c
@@ -15,15 +15,6 @@
 
 static struct hl_pll_cfg goya_pll_cfg[] = {
 	{
-		.name = "CPU", .base = mmCPU_PLL_BASE,
-		.nr = 2, .nf = 159, .od = 1, .nb = 79,
-		.div_factor = { 0, -1, 1, -1 },
-		.div_sel = { HL_PLL_PLL_CLK,
-			     HL_PLL_REF_CLK,
-			     HL_PLL_DIV_PLL,
-			     HL_PLL_REF_CLK
-		},
-	}, {
 		.name = "IC", .base = mmIC_PLL_BASE,
 		.nr = 0, .nf = 63, .od = 15, .nb = 31,
 		.div_factor = { 0, 1, 1, 2 },
diff --git a/goya/common.c b/goya/common.c
index 242b33f..f73b49d 100644
--- a/goya/common.c
+++ b/goya/common.c
@@ -171,11 +171,6 @@
 	return mmPSOC_GLOBAL_CONF_NON_RST_FLOPS_3;
 }
 
-u64 hl_pll_get_cpu_pll_base_off(void)
-{
-	return mmCPU_PLL_BASE;
-}
-
 int hl_pll_partial_init_get(void)
 {
 	return 0;
